File: golang/src/domain/frete.go
package domain

const distancia = 1000

type Frete struct {
	Valor float64
}

func (frete *Frete) AddProdutos(produto Produto, quantidade int) {
	if quantidade == 0 {
		quantidade = 1
	}
	frete.Valor += (distancia * produto.GetIndiceCalculoFrete()) * float64(quantidade)
}

func (frete Frete) GetValor() float64 {
	if frete.Valor > 0 && frete.Valor < 10 {
		return 10.0
	}

	return frete.Valor
}

File: node/src/domain/Frete.ts
import Produto from "./Produto"

export default class Frete{

    private distancia: number

    private valor: number;

    constructor(){
        this.distancia = 1000
        this.valor = 0
    }

    public addProdutos(produto: Produto, quantidade?: number){
        if (!quantidade) quantidade = 1
        if (produto)
            this.valor += (this.distancia * produto.getIndiceCalculoFrete()) * quantidade
        
    }

    public getValor(): number{
        return this.valor > 0 && this.valor < 10 ? 10.00 : parseFloat(this.valor.toFixed(2))
    }
}